AzureMonitorAgent/ama_tst/modules/helpers.py

```diff
@@ -93,16 +111,77 @@ def find_package_manager():
                 pkg_manager = "rpm"
             except subprocess.CalledProcessError:
                 pass
-
+    general_info['PKG_MANAGER'] = pkg_manager
```

Comment:
`general_info` is a global variable (since declared outside function scope). if you want to modify it within function scope, you need to declare `global general_info` at the top of the function. See [this page](https://www.w3schools.com/python/python_variables_global.asp)

Here's a simple example

```python
d = { "key": "pre" }

print(d["key"])

def change():
  # uncomment the below to ensure d actually changes
  # global d
  d = { "key": "post" }

change()

print(d["key"])
```